### PR TITLE
feat(mysql): add one command script to migrate on MySQL 8

### LINKED ISSUE #2414: [STORY-228] Document procedure to create Primary Keys and change db engine to innoDB
[STORY-228](https://notion.so/STORY-228)

For the migration of MySQL 5.7 to 8.x, we need to document an SQL Procedure to ease it for our customer.
In this procedure, we need to check if all tables check requirements for MySQL 8:
- [ ]  All tables must have `Primary Keys`
- [ ] All tables must use `InnoDB` engine

@curzolapierre already started this work and comment it here: https://github.com/Scalingo/appsdeck-database/issues/3035#issuecomment-1845049509
It handles the `Primary Keys` part but not the `InnoDB` engine yet.

### PR DESCRIPTION
**Pull request by curzolapierre**
Fixes #2414

### CODE REVIEW

**Review by ipfaze — CHANGES_REQUESTED** [review]
Some suggestions, overall LGTM :+1:

File: src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md [hunk 1]
```diff
@@ -69,3 +70,123 @@ In this situation, you need to add a primary key to the `foo` table. You need to
...
+
+## One Command Script
+
+Bellow you can find a script that can be run directly on the database to:
+- create missing primary keys to the tables concerned
+- set the engine to InnoDB for the tables concerned
+
+In fact, the script create two procedures, one for each previous actions.
+
+As always, we highly recommend to make a backup before executing the script.
```
Comment (yohann-bacha) [inline src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md:82]: **suggestion:** 

```suggestion
The following script creates two procedures, one for each previous actions. As always, we highly recommend to make a backup before executing the script.
```

**Review by yohann-bacha — CHANGES_REQUESTED** [review]
Some feedback on the english, but overall if the script works, that's great

File: src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md (hunk 1 above)
Comment (yohann-bacha) [reply src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md:82]: In fact, I prefer Mederic's suggestion on the warning block

**Review by ipfaze — APPROVED** [review]
LGTM :+1:

File: src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md [hunk 2]
```diff
@@ -69,3 +70,129 @@ In this situation, you need to add a primary key to the `foo` table. You need to
...
 ALTER TABLE 'foo' ADD COLUMN <column description> PRIMARY KEY;
 ```
+
+## One Command Script
+
+Below you can find a script that can be run directly on the database to:
+- create missing primary keys for the concerned tables
+- set the engine to InnoDB for the concerned tables
+
+In fact, the script create two procedures, one for each previous actions.
```
Comment (yohann-bacha) [inline src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md:80]: **suggestion:** I don't know if the "In fact" is really necessary here 

```suggestion
The script creates two procedures, one for each previous actions.
```

**Review by yohann-bacha — CHANGES_REQUESTED** [review]
One last nitpick but after this LGTM

File: src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md [hunk 3]
```diff
@@ -69,3 +70,129 @@ In this situation, you need to add a primary key to the `foo` table. You need to
...
+-- loop over all tables without PK
+alter_tables: LOOP
+
+    FETCH table_cursor INTO v_table;
+    IF v_finished = 1 THEN
+    LEAVE alter_tables;
+    END IF;
+
+    -- below the alter table which create PKs
+    SET @prepstmt = CONCAT('ALTER TABLE ',v_table,' ADD COLUMN id INT AUTO_INCREMENT NOT NULL PRIMARY KEY FIRST;');
```
Comment (yohann-bacha) [inline src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md:124]: **issue:** this column name is too common. for the record, I have a table that have a column named `id` but not primary. This query fails to execute.

```suggestion
    SET @prepstmt = CONCAT('ALTER TABLE ',v_table,' ADD COLUMN generated_pk INT AUTO_INCREMENT NOT NULL PRIMARY KEY FIRST;');
```

**Review by yohann-bacha — CHANGES_REQUESTED** [review]

**Review by yohann-bacha — APPROVED** [review]

**Pull request merged**